src/Handler/ErrorHandler/ErrorHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Mollie       https://www.mollie.nl
+ *
+ * @author      Mollie B.V. <info@mollie.nl>
+ * @copyright   Mollie B.V.
+ *
+ * @see        https://github.com/mollie/PrestaShop
+ *
+ * @license     https://github.com/mollie/PrestaShop/blob/master/LICENSE.md
+ * @codingStandardsIgnoreStart
+ */
+
+namespace Mollie\Handler\ErrorHandler;
+
+use Module;
+use Mollie;
+use Mollie\Config\Config;
+use MolliePrefix\Raven_Client;
+
+/**
+ * Handle Error.
+ */
+class ErrorHandler
+{
+	/**
+	 * @var Raven_Client
+	 */
+	protected $client;
+
+	/**
+	 * @var ErrorHandler
+	 */
+	private static $instance;
+
+	public function __construct()
+	{
+		/** @var Mollie */
+		$module = Module::getInstanceByName('mollie');
+
+		$this->client = new Raven_Client(
+			Config::SENTRY_KEY,
+			[
+				'level' => 'warning',
+				'tags' => [
+					'php_version' => phpversion(),
+					'ps_facebook_version' => $module->version,
+					'prestashop_version' => _PS_VERSION_,
+					'mollie_is_enabled' => \Module::isEnabled('mollie'),
+					'mollie_is_installed' => \Module::isInstalled('mollie'),
+				],
+			]
+		);
+
+		$this->client->install();
+	}
+
+	/**
+	 * @param \Exception $error
+	 * @param mixed $code
+	 * @param bool|null $throw
+	 *
+	 * @return void
+	 *
+	 * @throws \Exception
+	 */
+	public function handle($error, $code = null, $throw = true)
+	{
+		$this->client->captureException($error);
+		if ($code && true === $throw) {
+			http_response_code($code);
+			throw $error;
+		}
+	}
+
+	/**
+	 * @return ErrorHandler
+	 */
+	public static function getInstance()
```

Comment:
By using this function it kinda makes code untestable and requires adapter to be testable again, maybe (new ErrorHandler())->handle($error, $code, $throw) would be better or at least using DI to inject it into service